sim/dex-moves.ts

```diff
@@ -147,6 +147,10 @@ export interface MoveData extends EffectData, MoveEventMethods, HitEffect {
 	contestType?: string;
 	noPPBoosts?: boolean;
 
+	// stats used for the move
+	useBaseOffensiveStatAndBoosts?: ['target' | 'source', StatIDExceptHP];
+	useBaseDefensiveStatAndBoosts?: ['target' | 'source', StatIDExceptHP];
+
```

Comment:
These should be moved down to "Hit effect modifiers", where you deleted `defensiveCategory`.

Reply:
I wonder if `overrideOffensiveStat` and `overrideDefensiveStat` would be more readable.

Reply:
Yeah, I looked over everything. I think this would be much more readable if these were separate properties, instead of one combined property.

Reply:
One thing I might also be open to is a group:

```ts
overrideStats: {
  offensive: StatIDExceptHP,
  offensivePokemon: 'source' | 'target',
  defensive: StatIDExceptHP,
  defensivePokemon: 'source' | 'target',
};
```

I personally think grouping is unnecessary here, but if you like grouping, I wouldn't be _too_ opposed. Splitting into two tables is also a compromise, if you really want as minimal changes as possible.

Reply:
I want to try and communicate in some form that it doesn't use the opponent's items to boost stats since that isn't super obvious. overrideStats will have that problem as well. I do think splitting them is a good idea though. It will bulk things up a little, but increased readability is more important. I will have to make the pokemon value default to the source of the attack for offense and the target for defense.

I can see the appeal of grouping, but unlike conditions and secondary effects they won't rely on each other so I think it is ultimately unnecessary. The only benefit is it can help make the names shorter individually.

how about this as the names:
```
OverrideOffensiveStatPokemon: 'source' | 'target',
OverrideDefensiveStatPokemon: 'source' | 'target',
OverrideOffensiveStatAndBoosts: StatIDExceptHP,
OverrideDefensiveStatAndBoost: StatIDExceptHP
```
They aren't perfect as they don't exactly communicate that foul play would benefit from a choice band used by the attacker and not the defender but at that point, the changes to the variable names would be too much for it to be worth it. At the very least it should make it a bit more clear that choice band boosts body press.

Reply:
> I want to try and communicate in some form that it doesn't use the opponent's items to boost stats since that isn't super obvious.

I think that's too much information to fit in a variable name. I'd recommend a doc comment.

Reply:
I am going to push a change. Let me know what you think

Reply:
Looks great!

Reply:
Looks great!